src/DiagramNode.js

```diff
@@ -0,0 +1,237 @@
+            .append("rect")
+            .attr("x", 0)
+            .attr("y", 0)
+            .attr("width", this._styles.width)
+            .attr("height", this._styles.height)
+            .attr("fill", `url(#node-background-default-${this._node.name})`)
+            .attr("rx", 5)
+            .attr("stroke", NODE_STROKE_COLOR)
+            .attr("stroke-width", 1)
+            .attr("class", style['node-skeleton'])
+            .attr("id", this._node.name);
+    }
+
+    _renderIcon() {
+        this._icon = this.container.append("text")
+            .attr("x", NODE_GROUP_WIDTH/2)
+            .attr("y", (this._styles.height/2 + ICON_FONT_SIZE/2))
+            .attr("class", style['node-icon'])
+            .attr("font-family", this._iconFontFamily)
+            .attr("font-size", ICON_FONT_SIZE)
+            .attr("fill", NODE_ICON_COLOR)
+            .text(this._node.icon);
+    }
+
+    _renderLine() {
+        this._line = this.container
+            .append("line")
+            .attr("x1", NODE_GROUP_WIDTH)
+            .attr("y1", 0)
+            .attr("x2", NODE_GROUP_WIDTH)
+            .attr("y2", NODE_GROUP_WIDTH)
+            .attr("stroke", NODE_STROKE_COLOR)
+            .attr("stroke-width", 2)
+            .attr("opacity", 0);
+    }
+
+    _setNodeStyle() {
+        if (this._selected) {
+            this._setSpecialStyle(NODE_SELECTED_BG, 1);
+        } else if (this._highlighted) {
+            this._setSpecialStyle(this._groupColor, 0.4);
+        } else if (!this._highlightedSubsequent && (this._selectedMuted || this._highlightedMuted)) {
+            this._setMutedStyle();
+        } else {
+            this._setOriginalStyle();
+        }
+    }
+
+    _rendertNodeName() {
+        this._clipPath = this._defs
+            .append("clipPath")
+            .attr("id", "node-name-mask");
+
+        this._nodeNameMask = this._clipPath
+            .append("rect")
+            .attr("x", NODE_GROUP_WIDTH + 15)
+            .attr("y", 0)
+            .attr("width", this._nodeWidth - NODE_GROUP_WIDTH - 25)
+            .attr("height", this._styles.height);
+
+        this._nodeName = this.container
+            .append("g")
+            .attr("clip-path", "url(#node-name-mask)")
+            .append("text")
+            .text(this._node.name)
+            .attr("x", NODE_GROUP_WIDTH + 15)
+            .attr("y", (NODE_HEIGHT/2 + NODE_NAME_FONT_SIZE/2) - 1)
+            .attr("font-size", NODE_NAME_FONT_SIZE)
+            .attr("fill", NODE_NAME_COLOR)
+            .attr("class", style["node-name"]);
+    }
+
+    setStyle() {
+        this._setNodeStyle();
+    }
+
+    isSelected() {
+        return this._selected;
+    }
+
+    isSelectedSubsequent() {
+        return this._selectedSubsequent;
+    }
+
+    setSelected(value) {
+        this._selected = value;
+    }
+
+    setSelectedSubsequent(value) {
+        this._selectedSubsequent = value;
+    }
+
+    setHighlighted(value) {
```

Comment:
It's interesting, that these "public" methods like `setHighlighted` doesn't make any visual difference, until they're used together with `setStyle`. They just edit some internal property, am I right? Wouldn't it be better to "repaint" the node if I call some "public" method?

Reply:
Maybe I'm missing some basic context. If it's used "publicly" only inside `DiagramNodes` component and it's not meant to be used directly externally, then it's probably fine to leave it like it is.

Reply:
Exactly. it is public for purposes inside `DiagramNodes`.